cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -148,11 +148,11 @@ public synchronized void create(final NamespaceMeta metadata) throws Exception {
 
     // need to enforce on the principal id if impersonation is involved
     String ownerPrincipal = metadata.getConfig().getPrincipal();
-    if (ownerPrincipal != null) {
-      authorizationEnforcer.enforce(new KerberosPrincipalId(ownerPrincipal), authenticationContext.getPrincipal(),
-                                    Action.ADMIN);
+    Principal requestingUser = authenticationContext.getPrincipal();
+    if (ownerPrincipal != null && !ownerPrincipal.equals(requestingUser.getName())) {
```

Comment:
Let's move `!ownerPrincipal.equals(requestingUser.getName()` check to `DefaultAuthorizationEnforcer.enforce()`